frontend/src/components/teamsAndOrgs/teamSync.js

```diff
@@ -0,0 +1,482 @@
+  useOSMTeamModerators,
+  useOSMTeams,
+  useOSMTeamUsers
+} from '../../hooks/UseOSMTeams';
+import { UserAvatar } from '../user/avatar';
+import { OSM_TEAMS_API_URL } from '../../config';
+import { fetchExternalJSONAPI } from '../../network/genericJSONRequest';
+import { filterOSMTeamsMembers, getMembersDiff } from '../../utils/teamMembersDiff';
+import { joinTeamRequest, leaveTeamRequest } from '../../views/teams';
+import { Alert } from '../alert';
+
+const OSMTeamsLink = () =>
+  <a className="red link" target="_blank" rel="noopener noreferrer" href={OSM_TEAMS_API_URL}>
+    OSM Teams
+  </a>;
+
+
+const reSyncUsers = ({ tmTeamId, members, managers, osmTeamsId, osmteams_token, token, forceUpdate, setErrors }) => {
+  setErrors(false);
+  Promise.all([
+    fetchExternalJSONAPI(
+      new URL(`/api/teams/${osmTeamsId}/members`, OSM_TEAMS_API_URL),
+      `Bearer ${osmteams_token}`,
+      'GET',
+    ),
+    fetchExternalJSONAPI(
+      new URL(`/api/teams/${osmTeamsId}/moderators`, OSM_TEAMS_API_URL),
+      `Bearer ${osmteams_token}`,
+      'GET',
+    )
+  ]).then(([ osmTeamsUsers, osmTeamsModerators ]) => {
+    const { members: osmTeamsMembers, managers: osmTeamsManagers } = filterOSMTeamsMembers(
+      osmTeamsUsers.members.data,
+      osmTeamsModerators
+    );
+    const { usersAdded, usersRemoved } = getMembersDiff(
+      members,
+      osmTeamsMembers.map((user) => ({ username: user.name, function: 'MEMBER', active: true })),
+      false
+    );
+    const { usersAdded: managersAdded, usersRemoved: managersRemoved } = getMembersDiff(
+      managers,
+      osmTeamsManagers.map((user) => ({ username: user.name, function: 'MANAGER', active: true })),
+      true
+    );
+    const errors = [];
+    Promise.all([
+      ...managersRemoved.map((user) => leaveTeamRequest(tmTeamId, user, 'MANAGER', token)),
+      ...usersRemoved.map((user) => leaveTeamRequest(tmTeamId, user, 'MEMBER', token)),
+      ...managersAdded.map((user) =>
+        joinTeamRequest(tmTeamId, user, 'MANAGER', token)
+          .catch((e) => errors.push({username: user, function: 'MANAGER'}))
+        ),
+      ...usersAdded.map((user) =>
+        joinTeamRequest(tmTeamId, user, 'MEMBER', token)
+          .catch((e) => errors.push({username: user, function: 'MEMBER'}))
+        ),
+    ]);
+    setErrors(errors);
+    forceUpdate();
+  })
+};
+
+export const TeamSync = ({
+  osmTeamsId,
+  setOsmTeamsId,
+  setManagers,
+  setMembers,
+  managers,
+  members,
+  tmTeamId,
+  updateMode,
+  forceUpdate,
+  updateTeam,
+}) => {
+  const intl = useIntl();
+  const location = useLocation();
+  const params = location?.search ? new URLSearchParams(location.search) : null;
+  const osmteams_token = useSelector((state) => state.auth.osmteams_token);
+  const token = useSelector((state) => state.auth.token);
+  const [errors, setErrors] = useState(params?.get('syncUsersErrors'));
+  const [successModal, setSuccessModal] = useState(params?.get('access_token'));
```

Comment:
This state will make the popup show on every refresh if a user refreshes the page when the access_token URL param is available. Is this the intended behavior?

Reply:
I've improved this. After closing the modal, the `searchParams` will be updated, so the message will not be repeated if the user refreshes the page.